packages/world/src/requireNamespace.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { ResourceId, WorldResourceIdInstance } from "./WorldResourceId.sol";
+import { RESOURCE_NAMESPACE } from "./worldResourceTypes.sol";
+import { IWorldErrors } from "./IWorldErrors.sol";
+
+using WorldResourceIdInstance for ResourceId;
+
+/**
+ * @notice Checks if a given `resourceId` is a namespace.
+ * @dev Reverts with `IWorldErrors.World_InvalidResourceType` if the ID does not have the correct components.
+ * @param resourceId The resource ID to check.
+ */
+function requireNamespace(ResourceId resourceId) pure {
+  // Require the resourceId to have the namespace type
```

Comment:
This check with `resourceId.getNamespaceId()` feels a little "magical" since we optimised it in https://github.com/latticexyz/mud/pull/2142#discussion_r1457697540.

How about:

```suggestion
  // Require the resourceId to have the namespace type and the root name
  // (the resource ID is identical to the resource ID of its namespace)
```